internal/utils.go

```diff
@@ -408,18 +425,29 @@ func buildPolicyTemplate(
 		name = policyConf.Name
 	}
 
+	policyTemplateSpec := map[string]interface{}{
+		"object-templates":  policySpec,
+		"remediationAction": policyConf.RemediationAction,
+		"severity":          policyConf.Severity,
+	}
+
+	objectTemplatesRawContent, exists := policySpec[0]["object-templates-raw"]
+	if exists {
+		policyTemplateSpec = map[string]interface{}{
+			"object-templates-raw": objectTemplatesRawContent,
+			"remediationAction":    policyConf.RemediationAction,
+			"severity":             policyConf.Severity,
+		}
+	}
+
```

Comment:
What you have looks good, but I keep rethinking this. Building on what you have, perhaps if we pass in an `interface{}` for `objectTemplates` and then parse the type to set the key accordingly? Something like:

```golang
	policySpec := map[string]interface{}{}
	policySpec["remediationAction"] = policyConf.RemediationAction
	policySpec["severity"] = policyConf.Severity

	if _, ok := objectTemplates.(string); ok {
		policySpec["object-templates-raw"] = objectTemplates
	} else if _, ok := objectTemplates.([]map[string]interface{}); ok {
		policySpec["object-templates"] = objectTemplates
	}
```

Reply:
That does seem like it would be cleaner!